lib/location/location_core.c

```diff
@@ -506,19 +542,48 @@ static void location_core_periodic_work_fn(struct k_work *work)
 	location_core_location_get_pos(&current_config);
 }
 
-static void location_core_timeout_work_fn(struct k_work *work)
+static void location_core_method_timeout_work_fn(struct k_work *work)
 {
 	enum location_method current_method =
 		current_config.methods[current_method_index].method;
 
 	ARG_UNUSED(work);
 
-	LOG_WRN("Timeout occurred");
+	LOG_WRN("Method specific timeout expired");
 
 	location_method_api_get(current_method)->cancel();
 	location_core_event_cb_timeout();
 }
 
+static void location_core_timeout_work_fn(struct k_work *work)
+{
+	enum location_method current_method =
+		current_config.methods[current_method_index].method;
+
+	ARG_UNUSED(work);
+
+	LOG_WRN("Timeout for entire location request expired");
+
+	location_method_api_get(current_method)->cancel();
+	/* config->timeout needs to expire without fallbacks */
+
+	current_event_data.id = LOCATION_EVT_TIMEOUT;
+	event_handler(&current_event_data);
+
+	k_work_cancel_delayable(&location_core_timeout_work);
```

Comment:
Is this needed?

Reply:
This is copy-paste code and I have plans to get this go through the same handling in location_core_event_cb_fn as all other cases.

Reply:
Removed from this function.